src/lib.rs

```diff
@@ -101,6 +102,30 @@ struct OutputAssertion {
     fuzzy: bool,
 }
 
+#[derive(Debug, Copy, Clone)]
+enum OutputType {
+    StdOut,
+    StdErr,
+}
+
+impl OutputType {
+    fn select<'a>(&self, o: &'a Output) -> &'a [u8] {
+        match *self {
+            OutputType::StdOut => &o.stdout,
+            OutputType::StdErr => &o.stderr,
+        }
+    }
+}
+
+impl fmt::Display for OutputType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            OutputType::StdOut => write!(f, "stdout"),
+            OutputType::StdErr => write!(f, "stderr"),
+        }
+    }
+}
```

Comment:
Instead of doing this – an enum with two impl blocks – we could also use two unit structs and introduce a trait `OutputType` which both implement, that requires a Display impl and has a `select(&Output) -> &[u8]` method.

This way (as alluded to above) we can add a field `kind: T` to what is then `OutputAssertion<T>` and change the last two fields in `Assert` to `Option<OutputAssertion<Stdout>>` and `Option<OutputAssertion<Stderr>>` respectively.

I'm not sure this is a truly nicer design; I got the idea from wanting to encapsulate the whole output comparison business. Maybe it's totally over engineered 😄